modules/parquet/src/parquetjs/utils/decode-utils.ts

```diff
@@ -0,0 +1,327 @@
+// Forked from https://github.com/kbajalc/parquets under MIT license (Copyright (c) 2017 ironSource Ltd.)
+import {
+  ParquetCodec,
+  ParquetCompression,
+  ParquetData,
+  ParquetField,
+  ParquetType,
+  PrimitiveType,
+  SchemaDefinition
+} from '../schema/declare';
+import {CursorBuffer, ParquetCodecOptions, PARQUET_CODECS} from '../codecs';
+import {
+  ConvertedType,
+  Encoding,
+  FieldRepetitionType,
+  PageHeader,
+  PageType,
+  SchemaElement,
+  Type
+} from '../parquet-thrift';
+import {decompress} from '../compression';
+import {PARQUET_RDLVL_TYPE, PARQUET_RDLVL_ENCODING} from '../../constants';
+import {decodePageHeader, getThriftEnum, getBitWidth} from './read-utils';
+// import Fs = require('fs');
+
+/**
+ * Decode a consecutive array of data using one of the parquet encodings
+ */
+function decodeValues(
+  type: PrimitiveType,
+  encoding: ParquetCodec,
+  cursor: CursorBuffer,
+  count: number,
+  opts: ParquetCodecOptions
+): any[] {
+  if (!(encoding in PARQUET_CODECS)) {
+    throw new Error(`invalid encoding: ${encoding}`);
+  }
+  return PARQUET_CODECS[encoding].decodeValues(type, cursor, count, opts);
+}
+
+export async function decodeDataPages(
```

Comment:
- Generally I prefer to have exported functions at the top, with good JSDoc comments.
- This file is not just `utils` I feel - maybe `decoders.ts`?

Reply:
Done